src/Resources/config/http_kernel.xml

```diff
@@ -13,5 +13,19 @@
             <argument/>
             <argument type="service" id="logger" on-invalid="null"/>
         </service>
+
+        <!-- Overrides Symfony's security.exception_listener -->
+        <service id="security.exception_listener" class="Sonata\PageBundle\Listener\SecurityExceptionListener" abstract="true">
+            <tag name="monolog.logger" channel="security" />
+            <argument type="service" id="security.token_storage" />
+            <argument type="service" id="security.authentication.trust_resolver" />
+            <argument type="service" id="security.http_utils" />
+            <argument />
+            <argument type="service" id="security.authentication.entry_point" on-invalid="null" />
+            <argument>%security.access.denied_url%</argument>
+            <argument type="service" id="security.access.denied_handler" on-invalid="null" />
+            <argument type="service" id="logger" on-invalid="null" />
+            <argument>false</argument> <!-- Stateless -->
+        </service>
```

Comment:
An alternative would be service decoration instead of inheritance, but then you'd have to add all the public methods to the class :thinking: are there many of them?

Reply:
I personally have no experience with service decoration. Does this mean all the public methods with all functionality need to be copied to the decorating service? Or just the ones we're working on?

Reply:
All public methods signatures must be copied, but can act as proxy for the decorated object.